R/tokens_xptr.R

```diff
@@ -0,0 +1,209 @@
+#' @rdname tokens_xptr
+#' @export
+is.tokens_xptr <- function(x) {
+    identical(typeof(x), "externalptr") && "tokens_xptr" %in% class(x)
+}
+
+#' Convert a tokens object to a tokens_xptr object
+#'
+#' tokens_xptr is a external pointer object to process a large number of
+#' documents efficiently.
+#' @param x a [tokens] object to convert or a tokens_xptr object to
+#'   to deep-copy.
+#' @rdname tokens_xptr
+#' @export
+as.tokens_xptr <- function(x) {
+    UseMethod("as.tokens_xptr")
+}
+
+#' @rdname tokens_xptr
```

Comment:
We should collapse these into just `as.tokens_xtpr`, the way we do `as.tokens`, so there are not the object-specific methods in the package index.

Reply:
Done.